src/Parsers/Access/ParserGrantQuery.cpp

```diff
@@ -284,8 +283,49 @@ bool ParserGrantQuery::parseImpl(Pos & pos, ASTPtr & node, Expected & expected)
 
     AccessRightsElements elements;
     std::shared_ptr<ASTRolesOrUsersSet> roles;
-    if (!parseElementsWithoutOptions(pos, expected, elements) && !parseRoles(pos, expected, is_revoke, attach_mode, roles))
-        return false;
+
+    bool current_grants = false;
+    if (!is_revoke && ParserKeyword{"CURRENT GRANTS"}.ignore(pos, expected))
```

Comment:
Let's move this `parseKeyword{"CURRENT GRANTS"}` with the body of `if` into a separate function (because very long functions are difficult to read).

Reply:
(IMHO) it won't be much more readable to move the whole statement into a separate function, but the `if` body definitely should be a function